src/app/Table.elm

```diff
@@ -1,29 +1,95 @@
-module Table exposing (view, init, Msg)
+module Table exposing (view, init, initWithPlayersAndCup)
+
+{-|
+
+   This module handles the table of players,
+   and everything relating to that.
+
+-}
 
 import Html exposing (..)
+import Html.Events exposing (..)
+import Types exposing (PlayerID)
+import TableMsg exposing (Msg(..))
+import Dice
 
 
-{-
+type alias AppMsg =
+    Types.Msg
 
-   This module handles the view rendering for the table of players
-   and everything relating to that.
 
--}
+type alias PlayerName =
+    String
+
+
+type alias DiceCount =
+    Int
+
+
+type alias TablePlayer =
+    { id : PlayerID
+    , name : PlayerName
+    , diceCount : DiceCount
+    }
 
 
-type Model
-    = Model
+type alias TablePlayers =
+    List TablePlayer
 
 
-type Msg
-    = Msg
+type alias Model =
+    { tablePlayers : TablePlayers
+    , myCup : Dice.Cup
+    }
 
 
 init : Model
 init =
-    Model
+    { tablePlayers = noPlayers
+    , myCup = Dice.emptyCup
+    }
+
+
+initWithPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
+initWithPlayersAndCup tuples cup =
+    let
+        tupleToPlayer ( id, name, diceCount ) =
+            { id = id, name = name, diceCount = diceCount }
+    in
+        { tablePlayers = List.map tupleToPlayer tuples
+        , myCup = Dice.cupFromIntList cup
+        }
+
+
+noPlayers : TablePlayers
+noPlayers =
+    []
+
+
+testPlayers : TablePlayers
+testPlayers =
+    [ TablePlayer 1 "Joe" 6, TablePlayer 2 "Garrett" 5 ]
+
+
+view : (Msg -> AppMsg) -> Model -> Html AppMsg
+view privateMsg { tablePlayers } =
+    div []
+        [ button [ onClick <| privateMsg CallLiar ] [ text "Liar!" ]
```

Comment:
This illustrates how we call our `Table` module messages from within our `view` function. We use the passed in `privateMsg` function which wraps *our* `Msg` types in the `Types.Msg` type (which `AppMsg` is an alias for in this module).

That means the sub-modules are the ones that handle all of the "lifting" work done by the app, and yet we keep the ability to mount multiples in the app easily (by giving the second copy another, separate value constructor within the `AppMsg` type).

I wonder if we could gain the ability to compose these things hierarchically if we changed it to be `view : (Msg -> msg) -> Model -> Html msg`? That'd let us mount them hierarchically, possibly. I'll have to experiment with that.

WDYT?